### DISCUSSION TITLE
Question about this tool

### DISCUSSION THREAD

**Discussion by ghost** [quietvoid/dovi_tool#247] [Q&A] [answered]
Hello,

I'm interested in using your tool, but I'm not sure if it fits my requirement. Is it possible to extract RPU data in a FFmpeg compatible format so that I can pass them to my x265 or SVT-AV1 encoding job?

From my understanding, I need a binary RPU file I need to pass to FFmpeg along with some other parameters, please also see:

https://forum.doom9.org/showthread.php?t=176006

thanks in advance

**Comment by quietvoid** [comment, marked as answer]
You can use the `extract-rpu` subcommand to obtain a RPU file for x265.
Not too sure about SVT-AV1.

See https://github.com/quietvoid/dovi_tool#extract-rpu.
It's likely you would need to use a conversion mode depending on your Dolby Vision file.

**Reply by ghost** [reply]
Okay, but can I also choose to either get an xml or binary RPU ? I'm asking as the ffmpeg command at the very first post of the referenced forum thread uses xml?

From my understanding the RPU is a independent information besides the actual master-display metadata you have with normal HDR10. when it comes to x265 to SVT-AV HDR conversion, you just need to recalculate the vaules for each encoder . But I'm also not sure if SVT-AV1 has a DV RPU passing option

Thank you very much :)

**Reply by quietvoid** [reply]
> but can I also choose to either get an xml or binary RPU ?

No, you get a binary RPU. XML is not supported for encoding.
